main.go

```diff
@@ -217,26 +219,17 @@ func metricsServer(registry prometheus.Gatherer, host string, port int) {
 	log.Fatal(http.ListenAndServe(listenAddress, mux))
 }
 
-// registerCollectors creates and starts informers and initializes and
-// registers metrics for collection.
-func registerCollectors(registry prometheus.Registerer, kubeClient clientset.Interface, enabledCollectors options.CollectorSet, namespaces options.NamespaceList, opts *options.Options) {
-	informerFactories := []informers.SharedInformerFactory{}
-	for _, ns := range namespaces {
-		informerFactories = append(
-			informerFactories,
-			informers.NewSharedInformerFactoryWithOptions(
-				kubeClient, 0, informers.WithNamespace(ns),
-			),
-		)
-	}
-	activeCollectors := []string{}
-	for c := range enabledCollectors {
-		f, ok := kcollectors.AvailableCollectors[c]
-		if ok {
-			f(registry, informerFactories, opts)
-			activeCollectors = append(activeCollectors, c)
+type metricHandler struct {
+	c []kcollectors.Collector
+}
+
+func (m *metricHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	for _, c := range m.c {
+		for _, m := range c.Collect() {
+			_, err := fmt.Fprint(w, *m)
```

Comment:
I'm certain that we can still get significant speed up if we group all metrics from one object in a buffer and then just call `io.CopyBuffer` (or just `copy`) with a large enough buffer then we also don't waste any allocations here which look like they should be taking a lot of time themselves here.